tests/ui/ref_as_ptr.fixed

```diff
@@ -58,27 +64,27 @@ fn _msrv_1_75() {
     let mut_val = &mut 42_i32;
 
     // `std::ptr::from_{ref, mut}` was stabilized in 1.76. Do not lint this
-    let _ = val as *const i32;
-    let _ = mut_val as *mut i32;
```

Comment:
Didn't this change go way too far? `let _ = mut_val as *mut i32;` should absolutely trigger the lint, no? The issue isn't the temporariness of the resultant pointer, but whether the expression being cast is a `&x` or `&mut x` expression, right?